great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1564,6 +1564,9 @@ def list_datasources(self) -> List[dict]:
                 )
             )
             datasources.append(masked_config)
+
+        for datasource_name, datasource_config in self.config.fluent_datasources.items():
```

Comment:
```suggestion
        for datasource_name, datasource_config in self.fluent_datasources.items():
```
Excuse my ignorance here but why are we pulling from `self.config.fluent_datasources` instead of just `self.fluent_datasources`

I admit I don't know how or when `self.config.fluent_datasources` is set.

https://github.com/great-expectations/great_expectations/blob/4e9199122c4f9fb12b298d104b35ee8f4d1b0005/great_expectations/data_context/data_context/abstract_data_context.py#L4454-L4459

Reply:
`self.fluent_datasources` is a dict of hydrated business logic objects no? We need config because that's whats emitted to the user in `list_datasources`

Reply:
^ and the self.config.fluent_datasources is initialized in the 

[data_context/types/base.py]

(https://github.com/great-expectations/great_expectations/pull/7667/files#diff-2e0903f2ba2f628b8645e3bb6decf8d451c943e724fc3852f555e94e92b0430f)

Reply:
Maybe I just missed it but I've never written or reviewed any code that sets fluent datasource objects to `self.config`.
So I worry the test in this PR may be misleading for how this would actually work.

Reply:
hmm how do we actually persist fluent datasources? In YAML or in the Cloud DB right? Those values are provided back to us in the form of `DataContextConfig` no?

Reply:
yea when I tested this locally with a sqllite datasource (this FDS) was persisted in the YAML

```
import great_expectations as gx

context = gx.get_context()

context.datasources
# {}

context.sources.add_sqlite(name="my_datasource_1", connection_string="sqlite:///", kwargs={"isolation_level": "SERIALIZABLE"},)
# SqliteDatasource(type='sqlite', name='my_datasource_1', id=None, assets={}, connection_string=SqliteDsn('sqlite:///', scheme='sqlite', path='/'), create_temp_table=True, kwargs={'isolation_level': 'SERIALIZABLE'})

context.datasources
# {'my_datasource_1': SqliteDatasource(type='sqlite', name='my_datasource_1', id=None, assets={}, connection_string=SqliteDsn('sqlite:///', scheme='sqlite', path='/'), create_temp_table=True, kwargs={'isolation_level': 'SERIALIZABLE'})}

In [5]: context.list_datasources()
# []
```

Reply:
@cdkini 
>hmm how do we actually persist fluent datasources? In YAML or in the Cloud DB right? Those values are provided >back to us in the form of DataContextConfig no?

For cloud that might be true but for `FileDataContext` we use a separate code path to avoid the `instantiate_class_from_config` logic.

https://github.com/great-expectations/great_expectations/blob/4e9199122c4f9fb12b298d104b35ee8f4d1b0005/great_expectations/data_context/data_context/file_data_context.py#L199-L214